infrastructure/fargate-demo/template.yaml

```diff
@@ -171,14 +171,18 @@ Resources:
     Properties:
       VpcId: !Ref VPC
       CidrBlock: 10.0.1.0/24
-      AvailabilityZone: "eu-west-2a"
+      AvailabilityZone: !Select 
+        - 0
+        - Fn::GetAZs: !Ref 'AWS::Region'
 
   Subnet2:
     Type: AWS::EC2::Subnet
     Properties:
       VpcId: !Ref VPC
       CidrBlock: 10.0.2.0/24
-      AvailabilityZone: "eu-west-2b"
+      AvailabilityZone: !Select 
```

Comment:
Interesting, would be good to chat a bit more about this change?

Does this keep our app limited to any of the eu-west ones, or could it rock up on a us zone?
Wondering if we'll need to limit our stuff to particular zones, or if it's free to travel wherever there's availibility?

Reply:
Ah good question! The `samconfig.toml` file defines the region to deploy to, so if you run `sam deploy` these will always end up in `eu-west-2` (London). But it seems best practice (ie. there's a linter warning) is to pick up availability zones and regions programatically so we could deploy to a different region if we ever wanted to

Reply:
right, gotcha - that makes sense